src/features/activity/ActivityDetailsScreen.tsx

```diff
@@ -0,0 +1,297 @@
+            onMenuPress={onAddressItemPress(fromAccount)}
+          />
+        )}
+        {toAccount && (
+          <AddressActivityItem
+            accountAddress={toAccount}
+            marginHorizontal="l"
+            borderRadius="xl"
+            onMenuPress={onAddressItemPress(toAccount)}
+          />
+        )}
+      </Box>
+    )
+  }, [onAddressItemPress, transaction])
+
+  const title = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    if (confirmedSig.err) {
+      return t('activityScreen.transactionFailed')
+    }
+
+    const txnKey = `activityScreen.enrichedTransactionTypes.${enrichedTx.type}`
+    return i18n.exists(txnKey)
+      ? t(`activityScreen.enrichedTransactionTypes.${enrichedTx.type}`)
+      : t('activityScreen.enrichedTransactionTypes.UNKNOWN')
+  }, [i18n, t, transaction])
+
+  const description = useMemo(() => {
+    const enrichedTx = transaction as EnrichedTransaction
+    const confirmedSig = transaction as ConfirmedSignatureInfo
+
+    // Custom description that elipiizes the address
+    if (enrichedTx.description) {
+      const customDescription = enrichedTx.description
+        ?.split(' ')
+        .map((word) => {
+          // Remove addresses
+          if (solAddressIsValid(word)) {
+            return ''
+          }
+          return word
+        })
+        .join(' ')
+        .trim()
+
+      // capitalize first letter of description
+      return `${customDescription
+        .charAt(0)
+        .toUpperCase()}${customDescription.slice(1)}`
+    }
+
+    if (confirmedSig.err) {
+      return 'Error'
+    }
+
+    return 'Transaction Successful'
+  }, [transaction])
+
+  const handleOpenExplorer = useCallback(async () => {
+    const url = `https://explorer.solana.com/tx/${transaction.signature}?cluster=${cluster}`
+    // Open url in browser
+    await Linking.openURL(url)
+  }, [cluster, transaction.signature])
+
+  const toggleActionSheet = useCallback(
+    (open) => () => {
+      setOptionsOpen(open)
+    },
+    [],
+  )
+
+  const showToast = useCallback(() => {
+    if (!selectedAddress) return
+    Toast.show(
+      t('generic.copied', {
+        target: ellipsizeAddress(selectedAddress),
+      }),
+    )
+  }, [selectedAddress, t])
+
+  const handleCopyAddress = useCallback(() => {
+    if (!selectedAddress) return
+
+    Clipboard.setString(selectedAddress)
+    showToast()
+    triggerNavHaptic()
+    setOptionsOpen(false)
+  }, [selectedAddress, showToast, triggerNavHaptic])
+
+  const accountOptions = useCallback(
+    () => (
+      <>
+        <ListItem
+          key="copyAddress"
+          title={t('settings.sections.account.copyAddress')}
+          onPress={handleCopyAddress}
+          selected={false}
+        />
+      </>
+    ),
+    [handleCopyAddress, t],
+  )
+
+  return (
+    <Animated.View entering={DelayedFadeIn} style={globalStyles.container}>
```

Comment:
Just an FYI... We have a component `ReAnimatedBox` which could be used for stuff like this...
```ts
<ReAnimatedBox flex={1}>
{stuff}
</ReAnimatedBox>

Reply:
Done